src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -229,6 +229,7 @@ export class TerminalTaskSystem implements ITaskSystem {
 	}
 
 	public run(task: Task, resolver: ITaskResolver, trigger: string = Triggers.command): ITaskExecuteResult {
+		task = task.clone(); // A small amount of task state is stored in the task (instance) and tasks passed in to run may have that set already.
```

Comment:
Should it only do this conditionally if `instanceLimit > 1`?

Reply:
To fix this bug, yes it only needs to be done if `instanceLimit >= 1`. However, it would be better to not pollute the original tasks state regardless so I will keep it always cloning.